build/targets.mk

```diff
@@ -551,7 +616,7 @@ ifeq ($(TARGET),ANDROID)
   TARGET_CPPFLAGS += --sysroot=$(ANDROID_SYSROOT)
   TARGET_CPPFLAGS += -isystem $(ANDROID_SYSROOT)/usr/include/$(HOST_TRIPLET)
   TARGET_CPPFLAGS += -DANDROID
-  TARGET_CPPFLAGS += -D__ANDROID_API__=21
+  TARGET_CPPFLAGS += -D__ANDROID_API__=$(NDK_API)
```

Comment:
Making this a variable should be a separate commit

Reply:
Moved to separate commit within this series.

Reply:
I could have cherry-picked this one, but unfortunately, you made this depend on the other commits. Meanwhile, I had to implement a similar variable for v6.8.x to address a build problem. I suggest you rebase your PR on current master, fix the conflicts, and force-push.

Reply:
Sure will do. No problem